htdocs/content/crypto.php

```diff
@@ -193,75 +193,85 @@
-											$fres = dbquery("
-															SELECT
-																id
-															FROM
-																fleet
-															WHERE
-																entity_to=".$target->id()."");
-											if (mysql_num_rows($fres)>0)
+											if ($decryptlevel>=$cfg->value("crypto_number_of_fleets_level"))
 											{
-												if ($decryptlevel<0)
-												{
-													$out.="Es sind Flotten unterwegs\n";
-												}
-												else if ($decryptlevel<10)
-												{
-													$out.="Es sind ".mysql_num_rows($fres)." Flotten unterwegs\n";
-												}
-												else
+												$fres = dbquery("
+												SELECT
+													id
+												FROM
+													fleet
+												WHERE
+													entity_from=".$target->id()."
+													OR entity_to=".$target->id()."");
+												$out.="Es sind ".mysql_num_rows($fres)." Flotten unterwegs.\n\n";
+											}
+											else
+											{
+												$out.="Es sind Flotten unterwegs.\n\n";
+											}
+
+											if($decryptlevel>=$cfg->value("crypto_fleets_incoming_level"))
+											{
+												$out.="[b]Eintreffende Flotten[/b]\n\n";
+												$fres = dbquery("
+																SELECT
+																	id
+																FROM
+																	fleet
+																WHERE
+																	entity_to=".$target->id()."");
+												if (mysql_num_rows($fres)>0)
 												{
+													$out.="Es sind ".mysql_num_rows($fres)." Flotten unterwegs.\n\n";
 													while ($farr=mysql_fetch_row($fres))
 													{
 														$fd = new Fleet($farr[0]);
 														$source = $fd->getSource();
 														$owner = new User($fd->ownerId());
 
 														$out.='[b]Herkunft:[/b] '.$source.', [b]Besitzer:[/b] '.$owner;
-														$out.= "\n[b]Ankunft:[/b] ";
 
-														if ($decryptlevel<=15)
+														if ($decryptlevel>=$cfg->value("crypto_time_sec_level"))
 														{
-															$rand = random_int(0, 30*60*2);
-															$out.="Zwischen ".date("d.m.Y H:i",$fd->landTime() - $rand)." und ".date("d.m.Y H:i",$fd->landTime()+(2*30*60)-$rand)." Uhr";
+															$out.= "\n[b]Ankunft:[/b] ";
+															$out.=date("d.m.Y H:i:s",$fd->landTime())." Uhr";
 														}
-														elseif ($decryptlevel<=20)
+														elseif ($decryptlevel>=$cfg->value("crypto_time_min_level"))
 														{
-															$rand = random_int(0, 2*7*60);
-															$out.="Zwischen ".date("d.m.Y H:i",$fd->landTime()-$rand)." und ".date("d.m.Y H:i",$fd->landTime()+(2*7*60)-$rand)." Uhr";
+															$out.= "\n[b]Ankunft:[/b] ";
+															$out.=date("d.m.Y H:i",$fd->landTime())." Uhr";
 														}
-														elseif ($decryptlevel<=25)
+														elseif ($decryptlevel>=$cfg->value("crypto_time_15_level"))
 														{
-															$out.=date("d.m.Y H:i",$fd->landTime())." Uhr";
+															$rand = random_int(0, 2*7*60);
```

Comment:
This is a 14min timeframe but the config says 15

Reply:
15 is taken from the old code comments and the random modificator is unchanged. I don't know why there wass written +- 15 mins for 14. Just changing 2*7 to 15 would keep it an int.